### PR TITLE
Add support for ARCH parameter

### PR DESCRIPTION
**Pull request by zregvart**
Adds support for `ARCH` parameter which is passed to docker build as a `TARGETARCH` argument.

### CODE REVIEW

**Comment by pkubatrh** [comment]
Hi @zregvart , thanks for the PR.

Can you provide more info what this would be useful for? From what I can tell the build argument added by your patch is not used anywhere.

**Comment by zregvart** [comment]
@pkubatrh as it currently stands there is no way to build images for different architectures (say ARM64v8) other than changing the base image, this is passing a parameter so in the next PR like the one we did in https://github.com/Project31/s2i-base-container/pull/1 we can build images for other architectures.

I've written about this to the `sclorg@redhat.com` mailing list, I don't think the e-mail got through there seems to be mandatory moderation setup.

**Comment by pkubatrh** [comment]
Ah right, so the build argument is a means to provide a different namespace for the base image? I do not think this is something that would be useful upstream, sounds more like a change you would use in a fork (like you already have).

@hhorak WDYT? Also ping re: ML moderation issue mentioned above.

**Comment by praiskup** [comment]
[test]

File: common.mk [hunk 1]
```diff
@@ -59,7 +59,8 @@ script_env = \
 	CLEAN_AFTER=$(CLEAN_AFTER)                      \
 	DOCKER_BUILD_CONTEXT=$(DOCKER_BUILD_CONTEXT)    \
 	OPENSHIFT_NAMESPACES="$(OPENSHIFT_NAMESPACES)"  \
-	CUSTOM_REPO="$(CUSTOM_REPO)"
+	CUSTOM_REPO="$(CUSTOM_REPO)"                    \
+    ARCH="$(ARCH)"
```
Comment (praiskup) [inline common.mk]: Tiny: indent.

**Comment by hhorak** [comment]
> @hhorak WDYT? Also ping re: ML moderation issue mentioned above.

non-member contributions are approved by hand, I've done it now, sorry for delay.

**Comment by hhorak** [comment]
It looks like this PR is motivated by the way how the containers are built -- in our case (OSBS), we can use the same image names and thus even FROM argument for all arguments. I'm not sure how this will work for container built in CentOS Container pipeline, but I agree with @pkubatrh that it does not look very generic (suitable for this upstream).

**Comment by pkubatrh** [comment]
I am wondering... What is the reason you need to use the `$ARCH/` namespace in the `FROM` command? Afaict `docker.io/centos` and `docker.io/arm64v8/centos` are the same image on an aarch64 machine.

**Comment by zregvart** [comment]
@pkubatrh as far as I know there is no way to tell Docker which architecture the image is built for and which image to use from the manifest is dependant of the architecture of the machine the build is running on. So in order to force the build of an image for an architecture different from the architecture of the machine the image is built on we need to point to a specific image.

We can build ARM images on x86 machines [by using QEMU](https://medium.com/@kurt.stam/building-aarch64-arm-containers-on-dockerhub-d2d7c975215c).

**Comment by pkubatrh** [comment]
@zregvart Thanks for the info, did not think about the use case of building the arm images in QEMU. 

I would rather not make any changes to the existing Dockerfiles, especially some that could affect the tooling we are currently using (OSBS, CCCP, CWT, ...). I also vaguely remember there being some issues with build args in regards to the use of build cache last I experimented with them.

How about generating arch-specific Dockerfiles on demand? If you only need the `FROM` image's namespace to change it would be pretty easy to implement and I would be ok with that as long as the default behaviour does not change and the changes are limited to this repository (aside from submodule rebases of course).

Still, I think this is more of a feature that would be better suited for a fork, or maybe for the build tools themselves (docker/podman)?

**Comment by pkubatrh** [comment]
After a quick search a similar use case seems to be already tracked for buildah so you might want to join the discussion:
https://github.com/containers/buildah/issues/1590

**Comment by zregvart** [comment]
Moving this to the image build tooling makes perfect sense to me. This was really a way to get this done at the point when `buildx` wasn't available. I can see how this can complicate maintenance by affecting tooling taking this as input.
I'll close this and watch the space around `buildx` and suport in buildah on the issue you mentioned. Thank you for considering this.

**Pull request closed without merging**